sourcegraph/sourcegraph
# Remove insightIds from critical dashboard loading path

Closes https://github.com/sourcegraph/sourcegraph/issues/42263

## Background
This PR simply removes `insightIds` field from the dashboard GQL query; this information is no longer needed for the dashboard page's initial dashboard request (it was needed before because of the SettingCascade backend data source). It also removes one of the GQL backend fetch method and places the query right next to the UI component that uses this query.

**The outcome of this change**
All Insight dashboard response size was reduced from 12kb to 1.9kb

## Test plan
- Make sure that the dashboard page loads all insights (as it loads on the k8s)
- Check that the initial dashboard query doesn't load `insightIds` field
- Check that add/remove insight modal adds and/or removes insight from the dashboard properly 



## App preview:

- [Web](https://sg-web-vk-improve-dashboard-insight.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ikqwzkbvsm.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
